akka-http-core/src/test/scala/akka/http/impl/model/parser/HttpHeaderSpec.scala

```diff
@@ -631,6 +631,48 @@ class HttpHeaderSpec extends FreeSpec with Matchers {
           "1.2.3.4, akka.io\n          ^")
     }
 
+    "X-Forwarded-Host" in {
+      "X-Forwarded-Host: 1.2.3.4" =!= `X-Forwarded-Host`(remoteAddress("1.2.3.4"))
+      "X-Forwarded-Host: 1.2.3.4" <=!= `X-Forwarded-Host`(remoteAddress("1.2.3.4", Some(56789)))
+      "X-Forwarded-Host: 2001:db8:cafe:0:0:0:0:17" =!= `X-Forwarded-Host`(remoteAddress("2001:db8:cafe:0:0:0:0:17"))
+      "X-Forwarded-Host: 1234:5678:9abc:def1:2345:6789:abcd:ef00" =!= `X-Forwarded-Host`(remoteAddress("1234:5678:9abc:def1:2345:6789:abcd:ef00"))
+      "X-Forwarded-Host: 1234:567:9a:d:2:67:abc:ef00" =!= `X-Forwarded-Host`(remoteAddress("1234:567:9a:d:2:67:abc:ef00"))
+      "X-Forwarded-Host: 2001:db8:85a3::8a2e:370:7334" =!=> "2001:db8:85a3:0:0:8a2e:370:7334"
+      "X-Forwarded-Host: 1:2:3:4:5:6:7:8" =!= `X-Forwarded-Host`(remoteAddress("1:2:3:4:5:6:7:8"))
+      "X-Forwarded-Host: 1:2:3:4:5:6:7:8" <=!= `X-Forwarded-Host`(remoteAddress("1:2:3:4:5:6:7:8", Some(9)))
+      "X-Forwarded-Host: ::2:3:4:5:6:7:8" =!=> "0:2:3:4:5:6:7:8"
+      "X-Forwarded-Host: ::3:4:5:6:7:8" =!=> "0:0:3:4:5:6:7:8"
+      "X-Forwarded-Host: ::4:5:6:7:8" =!=> "0:0:0:4:5:6:7:8"
+      "X-Forwarded-Host: ::5:6:7:8" =!=> "0:0:0:0:5:6:7:8"
+      "X-Forwarded-Host: ::6:7:8" =!=> "0:0:0:0:0:6:7:8"
+      "X-Forwarded-Host: ::7:8" =!=> "0:0:0:0:0:0:7:8"
+      "X-Forwarded-Host: ::8" =!=> "0:0:0:0:0:0:0:8"
+      "X-Forwarded-Host: 1:2:3:4:5:6:7::" =!=> "1:2:3:4:5:6:7:0"
+      "X-Forwarded-Host: 1:2:3:4:5:6::" =!=> "1:2:3:4:5:6:0:0"
+      "X-Forwarded-Host: 1:2:3:4:5::" =!=> "1:2:3:4:5:0:0:0"
+      "X-Forwarded-Host: 1:2:3:4::" =!=> "1:2:3:4:0:0:0:0"
+      "X-Forwarded-Host: 1:2:3::" =!=> "1:2:3:0:0:0:0:0"
+      "X-Forwarded-Host: 1:2::" =!=> "1:2:0:0:0:0:0:0"
+      "X-Forwarded-Host: 1::" =!=> "1:0:0:0:0:0:0:0"
+      "X-Forwarded-Host: 1::3:4:5:6:7:8" =!=> "1:0:3:4:5:6:7:8"
+      "X-Forwarded-Host: 1:2::4:5:6:7:8" =!=> "1:2:0:4:5:6:7:8"
+      "X-Forwarded-Host: 1:2:3::5:6:7:8" =!=> "1:2:3:0:5:6:7:8"
+      "X-Forwarded-Host: 1:2:3:4::6:7:8" =!=> "1:2:3:4:0:6:7:8"
+      "X-Forwarded-Host: 1:2:3:4:5::7:8" =!=> "1:2:3:4:5:0:7:8"
+      "X-Forwarded-Host: 1:2:3:4:5:6::8" =!=> "1:2:3:4:5:6:0:8"
+      "X-Forwarded-Host: ::" =!=> "0:0:0:0:0:0:0:0"
+      "X-Forwarded-Host: akka.io" =!=
```

Comment:
The [spec](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host) suggests that domain names are allowed, it contains the example `id42.example-cdn.com`.

Reply:
Right!